### PR TITLE
Overhaul Mac layout

### PR DESCRIPTION
**Pull request by ermau**
This should address most spacing and sizing issues.

Before: 
screen shot 2019-02-25 at 4 46 03 pm

screen shot 2019-03-06 at 13 52 26

After:
screen shot 2019-03-06 at 13 53 28

screen shot 2019-03-06 at 13 37 18

### CODE REVIEW

File: Xamarin.PropertyEditing.Mac/Controls/BaseRectangleEditorControl.cs [hunk 1]
```diff
@@ -76,37 +76,37 @@ protected BaseRectangleEditorControl (IHostResourceProvider hostResources)
...
-				NSLayoutConstraint.Create (XEditor, NSLayoutAttribute.Top, NSLayoutRelation.Equal, this,  NSLayoutAttribute.Top, 1f, 0f),
+				NSLayoutConstraint.Create (XEditor, NSLayoutAttribute.Top, NSLayoutRelation.Equal, this,  NSLayoutAttribute.Top, 1f, 3f),
 				NSLayoutConstraint.Create (XEditor, NSLayoutAttribute.Left, NSLayoutRelation.Equal, this,  NSLayoutAttribute.Left, 1f, 0f),
 				NSLayoutConstraint.Create (XEditor, NSLayoutAttribute.Right, NSLayoutRelation.Equal, YEditor,  NSLayoutAttribute.Left, 1f, -10f),
-				NSLayoutConstraint.Create (XEditor, NSLayoutAttribute.Height, NSLayoutRelation.Equal, 1f, DefaultControlHeight),
+				NSLayoutConstraint.Create (XEditor, NSLayoutAttribute.Height, NSLayoutRelation.Equal, 1f, 18),
 
 				NSLayoutConstraint.Create (XLabel, NSLayoutAttribute.Top, NSLayoutRelation.Equal, XEditor,  NSLayoutAttribute.Bottom, 1f, -4f),
-				NSLayoutConstraint.Create (XLabel, NSLayoutAttribute.Height, NSLayoutRelation.Equal, 1f, DefaultControlHeight),
+				NSLayoutConstraint.Create (XLabel, NSLayoutAttribute.Height, NSLayoutRelation.Equal, 1f, 18),
```
Comment (netonjm) [inline Xamarin.PropertyEditing.Mac/Controls/BaseRectangleEditorControl.cs]: Constants ❤️

**Comment by vancura** [comment]
I am afraid we need a few more tweaks:

- [ ] Is it possible to move the checkbox `1pt` up so it aligns with the label?

image

- [ ] The gap in the middle should be `17pt`, it's `33pt` now:

image

Otherwise, it looks good.

Thanks!

**Comment by CartBlanche** [comment]
@vancura That **Property Search** hanging off the right still doesn't look right, even with these tweaks. Can we line up the right edge to be in line with the rest of the property controls??

**Comment by vancura** [comment]
> @vancura That Property Search hanging off the right still doesn't look right, even with these tweaks. Can we line up it's right edge to be in line with the rest of the property controls?

As I explained to Eric and Alan on a call this week, the whole toolbar would work well if it was used everywhere, as it [was designed](https://mseng.visualstudio.com/_git/DevDivDesign?path=%2Fspec%2Fvsmac-editor-toolbars%2FREADME.md&version=GBmaster&_a=preview). It will make more sense when it's `8pt` from the right. Here it looks _nearly_ like aligned with textfields, so it instead feels like a bug. When it's on the right side, our pattern flaws detecting brains won't suffer so much, I promise :)

File: Xamarin.PropertyEditing.Mac/Controls/Custom/SolidColorBrushEditor.cs [hunk 2]
```diff
@@ -79,12 +78,14 @@ private void Initialize (IHostResourceProvider hostResources)
 
 		private readonly CALayer background = new CALayer {
 			CornerRadius = 3,
-			BorderWidth = 1
+			BorderWidth = 1,
+			BorderColor = new CGColor (.5f, .5f, .5f, .5f),
```
[the remark this answers is not in the document]
Comment (ermau) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/SolidColorBrushEditor.cs]: It's just moved from another place, it's not new. It probably should be moved to theming.

**Pull request merged**